src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -802,29 +805,61 @@ public static decimal Round(decimal d, int decimals, MidpointRounding mode)
         public static double Round(double a)
         {
             // ************************************************************************************
-            // IMPORTANT: Do not change this implementation without also updating Math.Round(double),
+            // IMPORTANT: Do not change this implementation without also updating MathF.Round(float),
             //            FloatingPointUtils::round(double), and FloatingPointUtils::round(float)
             // ************************************************************************************
 
-            // If the number has no fractional part do nothing
-            // This shortcut is necessary to workaround precision loss in borderline cases on some platforms
+            // This is based on the 'Berkeley SoftFloat Release 3e' algorithm
```

Comment:
For reference: https://github.com/ucb-bar/berkeley-softfloat-3/blob/master/source/f64_roundToInt.c

The irrelevant support for the other rounding modes was not included.
Code comments were added for readability, they do not exist in the reference.

Reply:
It might be worth adding that note (i.e. that the other rounding modes from that algorithm are not included).